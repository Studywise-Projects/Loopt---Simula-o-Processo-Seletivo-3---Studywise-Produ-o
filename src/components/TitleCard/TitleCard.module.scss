@import '@/styles/_global.scss';

.card {
  display: flex;
  border-radius: 8px;
  box-shadow: 4px 4px 15px 0px rgba(0, 0, 0, 0.10);
  flex-wrap: wrap;
  justify-content: space-between;
  max-height: min-content;
  flex: 1;

  @media only screen and (min-width: 768px) {
    padding: 24px;
    width: auto;
  }
}

.content {
  display: flex;
  flex-direction: column;
  padding: 24px;

  @media only screen and (min-width: 768px) {
    width: 320px;
  }
}

.actions {
  display: flex;
  flex: 1;
  gap: 16px;
  padding: 8px 24px 24px 24px;
  flex-direction: row-reverse;
  
  @media only screen and (min-width: 768px) {
    flex-direction: row;
    justify-content: flex-end;
    padding: 24px;
    text-align: right;
  }
}

.actionsSimple {
  display: flex;
  flex: 1;
  gap: 16px;
  padding: 8px 24px 24px 24px;

  @media only screen and (min-width: 768px){
    display: none;
  }
}