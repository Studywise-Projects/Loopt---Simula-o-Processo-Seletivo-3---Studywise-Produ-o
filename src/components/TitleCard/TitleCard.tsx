import { Box, Card, CardActions } from '@mui/material';
import { Typography, Button, SearchBar } from '@/components';
import styles from './TitleCard.module.scss';

interface ITitleCard {
  variant?: 'withAction' | 'withoutAction';
  title: string;
  caption: string;
  actionButtonText: string;
  handleClick: () => void;
}

function TitleCard({
  variant = 'withAction',
  title,
  caption,
  actionButtonText,
  handleClick,
}: ITitleCard) {
  return (
    <>
      {variant === 'withAction' ? (
        <Card className={styles.card}>
          <Box className={styles.content}>
            <Typography variant='titleBlack' text={title} />
            <Typography variant='caption' text={caption} />
          </Box>
          <CardActions className={styles.actions}>
            <Button
              text={actionButtonText}
              handleClick={handleClick}
              isAuxButton={true}
            />
          </CardActions>
        </Card>
      ) : (
        <Card className={styles.card}>
          <Box className={styles.content}>
            <Typography variant='titleBlack' text={title} />
            <Typography variant='caption' text={caption} />
          </Box>
          <CardActions className={styles.actionsSimple}>
            <SearchBar
              isAuxSearch={true}
              id='search'
              label='Pesquisar'
              value={''}
              handleChange={() => undefined}
            />
          </CardActions>
        </Card>
      )}
    </>
  );
}

export default TitleCard;
