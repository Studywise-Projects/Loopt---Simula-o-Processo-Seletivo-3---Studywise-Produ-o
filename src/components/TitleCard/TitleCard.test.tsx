import { fireEvent, render } from '@testing-library/react';
import TitleCard from './TitleCard';

describe('Select', () => {
  it('Should match the snapshot', () => {
    const { getByText } = render(
      <TitleCard
        title='Desenvolvedor Front-End'
        caption='9 candidatos, 1 aprovado'
        actionButtonText='Embaralhar'
        handleClick={() => undefined}
      />,
    );

    const element = getByText('Desenvolvedor Front-End');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByText } = render(
      <TitleCard
        title='Desenvolvedor Front-End'
        caption='9 candidatos, 1 aprovado'
        actionButtonText='Embaralhar'
        handleClick={() => undefined}
      />,
    );

    const element = getByText('9 candidatos, 1 aprovado');
    expect(element).toBeInTheDocument();
  });

  it('Should call handleClickMock function once', () => {
    const handleClickMock = jest.fn();

    const { getByText } = render(
      <TitleCard
        title='Desenvolvedor Front-End'
        caption='9 candidatos, 1 aprovado'
        actionButtonText='Embaralhar'
        handleClick={handleClickMock}
      />,
    );
    const element = getByText('Embaralhar');
    fireEvent.click(element);

    expect(handleClickMock).toHaveBeenCalledTimes(1);
  });
});
