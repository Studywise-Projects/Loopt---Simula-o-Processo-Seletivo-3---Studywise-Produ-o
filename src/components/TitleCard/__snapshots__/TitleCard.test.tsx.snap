// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should match the snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 undefined css-ahj2mt-MuiTypography-root"
>
  Desenvolvedor Front-End
</p>
`;
